code/modules/reagents/reagent_containers/medspray.dm

```diff
@@ -91,3 +91,14 @@
 	name = "sterilizer spray"
 	desc = "Spray bottle loaded with non-toxic sterilizer. Useful in preparation for surgery."
 	list_reagents = list("sterilizine" = 60)
+/obj/item/reagent_containers/medspray/changed_name(newname)
+	if (lowertext(newname) == ("styptic powder"))
```

Comment:
don't constantly call lowertext

just do it once at the start of the proc
var/compare = lowertext(newname)
if(name == "styptic powder") 
        icon state

also don't call update icon every single time, just put it right at the bottom of the if chain once